v-next/hardhat-keystore/src/internal/tasks/set.ts

```diff
@@ -0,0 +1,61 @@
+import type { KeystoreLoader } from "../types.js";
+import type { UserInteractions } from "../ui/user-interactions.js";
+import type { HardhatRuntimeEnvironment } from "@ignored/hardhat-vnext/types/hre";
+import type { NewTaskActionFunction } from "@ignored/hardhat-vnext/types/tasks";
+
+import { setupDirectInterruptionsAndKeystoreLoader } from "../utils/setup-direct-interruptions-and-keystore-loader.js";
+import { validateKey } from "../utils/validate-key.js";
+
+interface TaskGetArguments {
+  key: string;
+  force: boolean;
+}
+
+const taskSet: NewTaskActionFunction<TaskGetArguments> = async (
+  setArgs,
+  hre: HardhatRuntimeEnvironment,
+): Promise<void> => {
+  const { keystoreLoader, interruptions } =
+    await setupDirectInterruptionsAndKeystoreLoader(hre);
+
+  await set(setArgs, keystoreLoader, interruptions);
+};
+
+export const set = async (
+  { key, force }: TaskGetArguments,
+  keystoreLoader: KeystoreLoader,
+  interruptions: UserInteractions,
+): Promise<void> => {
+  if (!(await validateKey(key))) {
+    process.exitCode = 1;
+    return interruptions.displayInvalidKeyErrorMessage(key);
+  }
+
+  // TODO: Bring in the setup password flow when implementing
+  // the encrypted version.
+  // await interruptions.setUpPassword();
```

Comment:
Note that this is also not an interruption, despite it being requesting the password. It still comes from the main task.